spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindNullDeref.java

```diff
@@ -1822,6 +1826,97 @@ private boolean isGeneratedCodeInCatchBlock(@NonNull ConstantPool cp, int pc) {
                     String exceptionName = cp.getConstantString(catchType, Const.CONSTANT_Class);
                     return "java/lang/Throwable".equals(exceptionName);
                 })
+                .collect(Collectors.toList());
+
+        ConstantPoolGen cpg = new ConstantPoolGen(cp);
+
+        LineNumberTable lineNumberTable = code.getLineNumberTable();
+        // The compiler generated code can also be written by the regular program. The only
+        // difference is that the line numbers for a lot of instructions are the same.
+        // This is what the code below relies on. Line numbers are optional, so it might
+        // not work in call cases.
+        //
+        // The following operations must have the same line numbers as the start or end of the original try catch block.
+        // - the reported code position (which is the null check)
+        // - at least one assignment
+        // - at least one call of addSuppressed
+        // - at least one throws statement
+        //
+        // The two calls to the close method need to have the same line number as the end of a throwables catch block.
+        // There must be also another catch of Throwable, but the line number does not need to match.
+        //
+        // To understand the code, please have a look at the test Issue600Test.
+        // TryWithResources* are the try-with-resources examples, ExplicitNullCheck* is TryWithResources* compiled and then decompiled.
+        if (lineNumberTable != null) {
```

Comment:
This method is getting quite long; perhaps move this block out to a helper?

Reply:
Done!